File: Chapter9_Example/src/main/java/DomainLogger.java
public class DomainLogger implements IDomainLogger {
    private final ILogger _logger;

    public DomainLogger(ILogger logger) {
        _logger = logger;
    }

    public void userTypeHasChanged(int userId, UserType oldType, UserType newType) {
        _logger.info("User " + userId + " changed type from " + oldType + " to " + newType);
    }
}

File: unit-testing/src/main/kotlin/book/chapter8/ambientcontext/DomainLogger.kt
package book.chapter8.ambientcontext

class DomainLogger(
    private val logger: ILogger,
) : IDomainLogger {

    override fun userTypeHasChanged(userId: Int, oldType: UserType, newType: UserType) {
        logger.info("User $userId changed type from $oldType to $newType")
    }

}